server/src/main/java/org/opensearch/index/mapper/ParametrizedFieldMapper.java

```diff
@@ -282,7 +282,7 @@ public Parameter<T> setValidator(Consumer<T> validator) {
         /**
          * Configure a custom serializer for this parameter
          */
-        protected Parameter<T> setSerializer(Serializer<T> serializer, Function<T, String> conflictSerializer) {
+        public Parameter<T> setSerializer(Serializer<T> serializer, Function<T, String> conflictSerializer) {
```

Comment:
Sadly this change broke k-NN plugin (https://github.com/opensearch-project/OpenSearch/pull/11182#discussion_r1396515318) so it is indeed a public API